docs/rules/curly/index.html

```diff
@@ -12,12 +12,47 @@
   In the code above, the author almost certainly meant for both `foo++` and `bar++`
   to be executed only if `foo === bar`. However, he forgot braces and `bar++` will be executed
   no matter what. This rule could prevent such a mistake.
-optionsDescription: Not configurable.
-options: null
+optionsDescription: |
+
+  The rule may be set to `true`, or to any combination of these arguments:
+
+  * `"all"` requires braces for all control-flow statements
+  * `"same-line-single-line"` requires braces for control-flow statements
```

Comment:
I can agree with this, though yeah, it pretty much blows a lot of my implementation away. I'll try not to get to caught up on any sunk-cost fallacy, though, since having a good implementation trumps however much time was spent creating a poor implementation.

Besides, at this point there appears to have been some changes made to `curlyRule.ts` that would cause a merge conflict. I haven't yet checked how big the changes are, so I might have to do a decent-sized refactor anyway.

My choices for my original implementation were basically because I was getting caught up on creating granular rules that we could build off of.

For instance, your "allow-same-line" and "allow-single-line" might be implemented with (or be an aliases for):

* allow-same-line: `[true, "next-line-single-line", "next-line-multi-line"]`
* allow-single-line: `[true, "same-line-single-line", "next-line-single-line"]`

The granular options I created definitely don't allow for the "never" cases, though. However, it wouldn't require a complete rewrite, and would basically just require more `if` statements inside of `validateStatementPositioning()` (along with the additional config constants, docs, strings, etc. -- of course.)

Anyway, food for thought. I'll clean some stuff up over the next few nights and update the pr.